### PR TITLE
[WIP] support redhat with cinder and glance enabled

### PR DESCRIPTION
**Pull request by chengtcli**
There are several workarounds for now, I have tagged them with `redhat-test` / `centos-test` comment.

### CODE REVIEW

**Comment by chengtcli** [comment]
I only focus on ceph and integration with cinder/glance, all other changes are hotfix/workaround. Code changes those are not related to ceph will be rollback and then do a rebase on PR #2363

**Comment by knandya** [comment]
@chengtcli : Thanks for the comments. @nirajdp76 Lets focus the review as per Cheng's comments

File: roles/ceph-common/defaults/main.yml [hunk 1]
```diff
@@ -1,33 +1,53 @@
...
   enabled: false
   #disks:
   #  - sdb
   #  - sdc
   #  - sdd
   #bcache_ssd_device: sdg 
-
-  version: 10.2.3~bbc1
+# ceph-centos-test
+  version: 10.2.3
```
Comment (knandya) [inline roles/ceph-common/defaults/main.yml]: Don't we need 10.2.3~bbc1 for Ubuntu?

File: roles/ceph-common/defaults/main.yml [hunk 2]
```diff
@@ -1,33 +1,53 @@
...
+      - libhugetlbfs
+      - libhugetlbfs-utils
+
   # Packages
-  debian_pkgs:
+  depend_pkgs:
     - python-pycurl
     - ntp
     - hdparm
-  ceph_pkgs:
```
Comment (knandya) [inline roles/ceph-common/defaults/main.yml]: No need for this list of packages?

File: roles/ceph-common/handlers/main.yml [hunk 3]
```diff
@@ -1,8 +1,16 @@
 ---
 - name: restart ceph mons
-  service: name=ceph-mon-all state=restarted
-  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool and not upgrade_ceph|bool
+  service: name={{ ceph.mon_service[os] }} state=restarted
```
Comment (knandya) [inline roles/ceph-common/handlers/main.yml]: What is [os] ? How does it get populated?

File: roles/cinder-common/templates/etc/cinder/cinder.conf [hunk 4]
```diff
@@ -19,8 +19,9 @@ state_path = {{ cinder.state_path }}
 rootwrap_config=/etc/cinder/rootwrap.conf
 api_paste_config = /etc/cinder/api-paste.ini
 
+# redhat-test
 {% if not ceph.enabled|bool -%}
-iscsi_helper=tgtadm
+iscsi_helper=targetcli
```
Comment (knandya) [inline roles/cinder-common/templates/etc/cinder/cinder.conf]: Will this change break Ubuntu?

Comment (chengtcli) [inline roles/cinder-common/templates/etc/cinder/cinder.conf]: Yea, good catch.

File: roles/ceph-common/defaults/main.yml (hunk 1 above)
Comment (chengtcli) [inline roles/ceph-common/defaults/main.yml]: Yes, we need. It's in the coming commit

File: roles/ceph-common/defaults/main.yml (hunk 2 above)
Comment (chengtcli) [inline roles/ceph-common/defaults/main.yml]: Yes, install packages in task directly, without variable.

File: roles/ceph-common/handlers/main.yml (hunk 3 above)
Comment (chengtcli) [inline roles/ceph-common/handlers/main.yml]: `os` is ubuntu or rhel, it's defined in precheck

**Comment by knandya** [comment]
@chengtcli PR #2363  is merged.. Can you cleanup the PR 2364?

**Comment by chengtcli** [comment]
@knandya , sure.

**Comment by chengtcli** [comment]
A new PR #2387  is issued, closing this PR...

**Pull request closed without merging**